src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,365 @@
+  - Mutually exclusive with `static`.
+  - Cannot be used if an explicit `this` parameter is present.
+
+```cpp
+auto next = [i = 0] mutable { return i++; };
+
+next() // 0
+next() // 1
+next() // 2
+```
+
+These may be followed by a `noexcept` specifier, to determine whether calling the lambda may throw an exception.
+
+&nbsp;
+
+### Template parameters
+
+A lambda's `operator()` may be templated to accept template parameters since C++20:
+
+```cpp
+auto lambda = []<typename T>(const T& x) {
+	return x;
+};
+
+// Deduce template argument from function argument
+lambda(5);
+
+// Pass template argument explicitly
+lambda.template operator()<double>(5);
+```
+
+&nbsp;
+
+### Attributes
+
+Lambdas may be given attributes that apply to their `operator()`s since C++23:
+
+```cpp
+auto very_important_number = [][[nodiscard]] { return 4; };
+```
+
+These attributes are placed right after (optional) template parameters.
+
+For details, see [cppreference](https://en.cppreference.com/w/cpp/language/lambda).
+
+&nbsp;
+
+&nbsp;
+
+## Notes
+
+### Inheritance
+
+Lambdas can be derived from:
+
+```cpp
+auto base = [] { std::puts("Hello, world!"); };
+
+struct : decltype(base) {} derived;
+
+derived(); // prints "Hello, world!"
+```
+
+This, in combination with multiple inheritance, allows for a very neat "visitor" class:
+
+```cpp
+template<typename... Fs>
+struct visitor : Fs... {
+	using Fs::operator()...;
+};
+
+visitor {
+	[](int) { std::puts("int"); },
+	[](double) { std::puts("double"); },
+	[](...) { std::puts("unknown"); }
+}(5); // prints "int"
+```
+
+&nbsp;
+
+### Capturing function parameters
+
+While you can use a function's parameters in its `noexcept` specifier and trailing `requires` clause, using a lambda
+there which captures the function's parameters is invalid:
+
+```cpp
+// Fine
+void f(int x) noexcept(noexcept(x)) {}
+
+// Invalid
+void f(int x) noexcept(noexcept([x] { x; })) {}
+```
+
+This is because the lambda is technically not in function or class scope, and thus cannot have captures. See
+[expr.prim.lambda.capture#3](https://timsong-cpp.github.io/cppwp/n4950/expr.prim.lambda.capture#3). This can sometimes
+be worked around using a `requires` expression:
+
+```cpp
+void f(int x) noexcept(requires(decltype(x) x) {
+	requires noexcept([x] { x; });
+}) {}
+```
+
+&nbsp;
+
+### Type aliases
+
+Making a type alias with an in-line lambda in a header file or module interface violates the
+[One-Definition Rule](https://en.cppreference.com/w/cpp/language/definition) because aliases are not a "definable item",
+so lambdas in them are not allowed to match with other lambda declarations in other
+[translation units](https://en.cppreference.com/w/cpp/language/translation_phases#Translation_process):
+
+```cpp
+// Bad
+using T = decltype([] {});
+
+// Bad
+template<auto> struct A {};
+using T = A<[] {}>;
+
+// Ok
+auto lambda = [] {};
+using T = decltype(lambda);
+```
+
+&nbsp;
+
+### In-line partial specialization
```

Comment:
I'm afraid I don't see why this is useful. It's also not really a specialization, which can be most clearly seen in the index sequence example. I would recommend keeping that example since that is a common useful pattern.